memory/src/main/scala/filodb.memory/MemFactory.scala

```diff
@@ -318,10 +325,12 @@ class BlockMemFactory(blockStore: BlockManager,
       }
     } else {
       val newBlock = requestBlock()
-      if (markFullBlocksAsReclaimable) {
-        block.markReclaimable()
-      } else {
-        fullBlocks += block
+      if (!metadataSpanActive) {
```

Comment:
My main Q is, here, `metadataSpanActive` will always be true because we always allocate chunks in the middle of an span ie between startMetadata and endMetadata.   Thus the code in there will effectively not be called.... while the foreach loop in endMeta seems to mark things as reclaimable, I'm wondering if we are missing any cases by effectively not calling the above.  Also if we aren't really going to use that code might as well comment it out.

Reply:
There's already a check below for if (metadataSpanActive), so I assumed both cases are possible.

Reply:
It is possible to allocate memory directly without a span.